packages/snaps-execution-environments/src/common/endowments/date.ts

```diff
@@ -0,0 +1,54 @@
+import { rootRealmGlobal } from '../globalObject';
+
+const MAXIMUM_NOISE = 5;
+
+/**
+ * Creates a {@link Date} constructor, with most of the same properties as the global object.
+ * The Date.now() function has added noise as to limit its precision and prevent potential timing attacks.
+ * The Date constructor uses this now() function to seed itself if no arguments are given to the constructor.
+ *
+ * @returns A modified {@link Date} constructor with limited precision.
+ */
+function createDate() {
+  const keys = Object.getOwnPropertyNames(
+    rootRealmGlobal.Date,
+  ) as (keyof typeof Date)[];
+
+  let currentTime = 0;
+  const now = () => {
+    const actual = rootRealmGlobal.Date.now();
+    const noise =
+      rootRealmGlobal.crypto.getRandomValues(new Uint32Array(1))[0] %
```

Comment:
Can you move this to a util function, as this is used in the math endowment as well?

Reply:
Not relevant in the current iteration